orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/instance/WaitForUpInstancesTask.groovy

```diff
@@ -154,4 +167,38 @@ class WaitForUpInstancesTask extends AbstractWaitingForInstancesTask {
     // no health indicators is indicative of being down
     return !healths || healths.any { it.state == 'Down' || it.state == 'OutOfService' }
   }
+
+  private static Map<String, Integer> getServerGroupCapacity(Stage stage, Map serverGroup) {
+    def serverGroupCapacity = serverGroup.capacity as Map<String, Integer>
+
+    def initialTargetCapacity = getInitialTargetCapacity(stage, serverGroup)
+    if (!initialTargetCapacity) {
+      return serverGroupCapacity
+    }
+
+    if (serverGroup.capacity.max == 0 && initialTargetCapacity.max != 0) {
+      serverGroupCapacity = initialTargetCapacity
+    }
+
+    return serverGroupCapacity
+  }
+
+  private static Map<String, Integer> getInitialTargetCapacity(Stage stage, Map serverGroup) {
+    def katoTasks = (stage.context."kato.tasks" as List<Map<String, Object>>)?.reverse()
```

Comment:
Super clumsy to fetch "kato" results ... same comment as above re: edge case testing.